Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/StatusBean.java

```diff
@@ -180,4 +203,37 @@ private void initPieChart() {
         refreshPieChart();
     }
 
+    public List<CertificateDTO> getCertificateInfo() {
+        try {
+            certificateList = new ArrayList<>();
+            addCertificates(certificateService.fetchKeyStores(), certificateList);
+            addCertificates(certificateService.fetchTrustStores(), certificateList);
+
+            Collections.sort(certificateList, new Comparator<CertificateDTO>() {
+                @Override
+                public int compare(CertificateDTO o1, CertificateDTO o2) {
+                    return o2.getExpiresInDays() > o1.getExpiresInDays() ? -1
+                            : o2.getExpiresInDays() == o1.getExpiresInDays() ? 0 : 1;
+                }
+            });
+        } catch (CertificateManagerException e) {
+            LOG.error("Unable to get certificate details {}", e.getLocalizedMessage(), e);
```

Comment:
Should we return an empty list here if the method fails? As it stands, we could technically get a partially populated list (also unordered) if fetching the Keystore is successful, but the Trust store throws the exception.

Reply:
sure, addressing with next PR.